content/Link Tracking with StimulusReflex - Part II.md

```diff
@@ -0,0 +1,613 @@
+# Link Tracking with StimulusReflex - Part II
+
+In the [first part of this series](TODO: Add link to first part) we took care of setting up our project to use Stimulus Reflex. Now it's time to get our hands dirty and write some code. The first thing we are going to do is take care of managing our shortened links.
+
+![Header image](images/stimulus_reflex.jpeg)
+
+## Creating the model
+
+Let's start with some good old fashioned Rails code, we'll create the migration for our links and the associated model:
+
+
+```ruby
+class CreateShortenedLinks < ActiveRecord::Migration[6.0]
+  def change
+    create_table :shortened_links do |t|
+      t.string :name, null: false
+      t.string :shortened_path, null: false
+      t.string :original_url, null: false
+      t.integer :views_count, default: 0
+
+      t.timestamps
+    end
+  end
+end
+```
+
+```ruby
+# app/models/shortened_link
+
+class ShortenedLink < ApplicationRecord
+  validates :name, :shortened_path, :original_url, presence: true
+  validates :shortened_path, uniqueness: true
+end
+```
+
+Notice how we added some simple validations to make it more fun.
```

Comment:
Wouldn't this add noise to the actual goal of the tutorial?

I would remove all DB constraints (null: false, default: '') and also the model validations just to keep the article focused  on one thing: Stimulus Reflex

Reply:
ohhh nevermind I just saw these validations are being used below in the example! Simply amazing!